distributed/worker.py

```diff
@@ -2532,7 +2532,7 @@ async def handle_missing_dep(self, *deps, **kwargs):
                     logger.debug("New workers found for %s", dep.key)
                     self.log.append((dep.key, "new workers found"))
                     for dependent in dep.dependents:
-                        if dependent.key in dep.waiting_for_data:
+                        if dep.key in dependent.waiting_for_data:
```

Comment:
I think I will plainly reject PRs which suggest using `dep` as a variable name in the future 🤔 

BTW this `if` clause in general seems a bit weird. there is no actual reason why one should not reschedule the task. however, I am not entirely sure but I believe this is here to protect `Worker.data_needed` from overflowing. We should consider replacing this with an ordered-set-like thingy (e.g. a dict) or some other data structure. Ideally we'd like to have a priority sorted set-like iterator here but well... that's a problem for another day